adr/0251-use-catalyst-for-client-side-behaviours.md

```diff
@@ -0,0 +1,28 @@
+# 251. Use Catalyst for client-side behaviours
+
+Date: 2022-04-12
+
+## Status
+
+Accepted
+
+## Context
+
+<!-- The issue motivating this decision, and any context that influences or constrains the decision. -->
+
+Existing Primer ViewComponents behaviours don't use [Catalyst](https://github.github.io/catalyst) or any other dependency, while GitHub developers often start writing a component using Catalyst.
+
+Removing the Catalyst dependency from the Web Component is an extra burden on Web Developers with few benefits. The library is ~1kb of JavaScript, and github.com already includes the library as part of its client-side bundle. Adding the library to Primer ViewComponents does not affect github.com and has a minimal effect on other applications using Primer ViewComponents.
+
+## Decision
```

Comment:
Most ADR's I've seen usually include alternatives considered listing out pros and cons of each approach. Thoughts on adding something similar here?

Reply:
Happy to add any thing here but didn't see anything like that in other ADRs for this repo or in the linked reading. I feel like anything I'd come up with would be arbitrary straw-man arguments.

Maybe I could add not using Catalyst and using other similar libraries, would that work here?

Reply:
We've done the pro/con approach in a good number of ADR's, like [Global Notices](https://github.com/github/fast/blob/main/adr/0000-storage-backing-global-notice-next.md) and [snek](https://github.com/github/snek/blob/main/docs/adrs/04-ownership-reporting-architecture.md). Memex does [something similar](https://github.com/github/memex/blob/main/docs/adr/adr-022-api-convention-in-the-frontend.md#alternatives-explored) via `Alternatives Explored`.

> Maybe I could add not using Catalyst and using other similar libraries, would that work here?

I think that would be perfect 👍. 

I view ADR's mostly as tools to help drive folks towards consensus to one solution over other solutions as well as being an artifact to give readers context on why a specific approach was chosen compared to alternatives. If y'all are using them differently here that's totally fine, and feel free to consider my comments non-blocking. 🙂